py/makeversionhdr.py

```diff
@@ -57,7 +57,7 @@ def get_version_info_from_git():
     return git_tag, git_hash, ver
 
 def get_version_info_from_docs_conf():
-    with open("%s/docs/conf.py" % sys.argv[0].rsplit("/", 2)[0]) as f:
+    with open(os.path.join(os.path.dirname(sys.argv[0]), "docs", "conf.py")) as f:
```

Comment:
Likely the build succeeded because ```get_version_info_from_docs_conf``` is only called if building 'bare' uPy sources and not when building git sources. The latter is the case probably since you're creating PRs..
Cleaning does not delete generated headers because I never bothered implementing that; I don't think it is needed currently: makeqstrdata.py and makeversionhdr.py are called for every build.